extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,199 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+const productImagesValidationConfig = {
+  'url': [validateRequired, validateURLs],
+  'alt': [validateRequired, validateString],
+};
+
+/** @const {!Object<!Object<string, !Array<function>>>} */
+const aggregateRatingValidationConfig = {
+  'ratingValue': [validateRequired, validateNumber],
+  'reviewCount': [validateRequired, validateNumber],
+  'reviewUrl': [validateRequired, validateURLs],
+};
+
+/** @const {!Object<!Object<string, !Array<function>>>} */
+export const productValidationConfig = {
+  /* Required Attrs */
+  'productUrl': [validateRequired, validateURLs],
+  'productId': [validateRequired, validateString, validateHTMLId],
+  'productTitle': [validateRequired, validateString],
+  'productBrand': [validateRequired, validateString],
+  'productPrice': [validateRequired, validateNumber],
+  'productImages': [
+    validateRequired,
+    createValidateConfigArray(productImagesValidationConfig),
+  ],
+  'productPriceCurrency': [validateRequired, validateString, validateCurrency],
+  /* Optional Attrs */
+  'aggregateRating': [
+    createValidateConfigObject(aggregateRatingValidationConfig),
+  ],
+  'productIcon': [validateURLs],
+  'productTagText': [validateString],
+};
+
 /**
  * @typedef {{
  *  items: !Array<!ShoppingConfigDataDef>,
  * }}
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Validates an Object using the validateConfig function.
+ * @param {?Object=} validation
```

Comment:
`@param {!Object<string, !Array<function>>} config`, just like the type of the validation configs defined at the top of the file

Reply:
Okay, but for `getObjectArrayValidationFnForConfig`, it's an array of objects, so the type should be `@param {!Array<!Object<string, !Array<function>>>} config`